### PR TITLE
Docs: update OpenStack documentation

### PR DESCRIPTION
**Pull request by Fedosin**
This PR updates OpenStack documentation based on recent changes.

1. Updates Swift requirements. Now Swift is not a mandatory service but just an optional backend for image registry.
2. Adds Glance requirements. Glance is used for storing RHCOS images and bootstrap ignition configs.
3. Adds RHCOS image customization section for disconnected environments.
4. Moves "Cluster destruction if its metadata has been lost" section from "known-issues" to "troubleshooting".

### CODE REVIEW

File: docs/user/openstack/customization.md [hunk 1]
```diff
@@ -1,8 +1,22 @@
-# OpenStack Platform Customization
+# Installation customizations
+
+This document describes the ways of possible installation customizations.
+
+## Table of Contents
+
+- [Installation customizations](#installation-customizations)
+  - [OpenStack Platform Customization](#openStack-platform-customization)
```
Comment (mandre) [inline docs/user/openstack/customization.md]: You left an uppercase `S` in there, it can't find the anchor.

Comment (Fedosin) [reply docs/user/openstack/customization.md]: done :)

**Comment by iamemilio** [comment]
Are we supposed to be exposing the image overrides to the customer? My understanding was that these are not supported, and should not be recommended to customers.

**Comment by Fedosin** [comment]
@iamemilio I did this because the way we use the variable is a little bit unusual and I got a lot of PMs where people ask to explain how this works. So I wanted to have it documented somewhere and give them the link instead of explaining it over and over again.
Also, you can't install OpenShift on a disconnected environment without setting this variable, so users should be aware of this possibility.

**Comment by iamemilio** [comment]
Hmm that does seem problematic, however I think I would like alignment from @sdodson before approving this.

**Comment by sdodson** [comment]
/approve
I'm fine with the custom image stuff being in dev docs. Don't know if you wanted to squash any of this down or not.

**Pull request merged**